api/v1alpha1/provisioning_validation.go

```diff
@@ -0,0 +1,205 @@
+package v1alpha1
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var (
+	log = ctrl.Log.WithName("provisioning_validation")
+)
+
+// Validate the contents of the provisioning resource
+/*
+   Managed:
+   "ProvisioningInterface"
+   "ProvisioningIP"
+   "ProvisioningNetworkCIDR"
+   "ProvisioningDHCPRange"
+   "ProvisioningOSDownloadURL"
+
+   Unmanaged:
+   "ProvisioningInterface"
+   "ProvisioningIP"
+   "ProvisioningNetworkCIDR"
+   "ProvisioningOSDownloadURL"
+
+   Disabled:
+   "ProvisioningIP"
```

Comment:
This isn't accurate.

Disabled is BOTH ProvisioningIP & ProvisioningNetworkCIDR or neither.

Reply:
oh dangit, got the wrong ones. thanks Stephen

Reply:
fixed.  Too fast with the vim-fu there

Reply:
oh, both or neither!  ah I didn't realize.  I'll have to do another bit of dev then.

Reply:
Thanks, looks like the current version of this is correct